app/mailers/producer_mailer.rb

```diff
@@ -0,0 +1,37 @@
+
+class ProducerMailer < Spree::BaseMailer
+
+  def order_cycle_report(producer, order_cycle)
+    @producer = producer
+    @coordinator = order_cycle.coordinator
+    @order_cycle = order_cycle
+
+    subject = "[#{Spree::Config[:site_name]}] Order cycle report"
+
+    # if @order_cycle.distributors.any?
+    #   first_producer = @order_cycle.distributors.first
+    #   @distribution_date = @order_cycle.pickup_time_for first_producer
+    #   subject += " for #{@distribution_date}" if @distribution_date.present?
+    # end
+
+    # TODO: what if producer handling orders into multiple order cycles?
+    @orders = Spree::Order.complete.not_state(:canceled).managed_by(@producer.owner)
+    # puts @orders.size
+
+    # Create a single flat list of all line items
+    @line_items = @orders.map(&:line_items).flatten
```

Comment:
This will create an n+1 queries problem - for every order, we'll do a separate SQL query to fetch the line items, and then combine the list in ruby. We can fetch the whole lot in one go like this:

```ruby
Spree::LineItem.where(order_id: @orders)
```